firebase-appdistribution/src/test/java/com/google/firebase/appdistribution/impl/ApkUpdaterTest.java

```diff
@@ -176,14 +183,23 @@ public void updateApk_whenSuccessfullyUpdated_notificationsSetCorrectly()
     doReturn(new ByteArrayInputStream(TEST_FILE.getBytes()))
         .when(mockHttpsUrlConnection)
         .getInputStream();
+    // Sleep thread to simulate actually making the request, which gives us time to add
+    // a listener to the returned UpdateTask in time to get all the progress updates
+    when(mockHttpsUrlConnectionFactory.openConnection(TEST_URL))
+        .thenAnswer(
+            invocation -> {
+              try {
+                Thread.sleep(100);
```

Comment:
Could you use a `CountDownLatch` instead? So you'd have full control when this invocation completes.

Reply:
Yeah I like that. Updated.